dev/com.ibm.ws.dynamic.bundle/src/com/ibm/ws/dynamic/bundle/BundleFactory.java

```diff
@@ -208,19 +208,26 @@ private void computeExtrasSha() {
         }
     }
 
-    private static MessageDigest shaDigest;
+    private static String shaDigestAlgorithm;
```

Comment:
I think some more comments should be added here to explain why the code is crafted in this way. It isn't clear and obvious, particularly since this is partly to workaround an OpenJDK bug.

Reply:
Thanks, @tevans78. I added comments indicating the SHA algorithm remains fixed over the life of the server and is set during the first usage of BundleFactory. During checkpoint the preferred algorithm is not available, so BundleFactory defaults to a known available algorithm. The default algorithm is used in both the checkpoint and restored server processes.

The comment does not address the OpenJDK bug exposed by the build break. The break was caused by a change in the use of MessageDigest which allowed multiple threads to access/share the same MessageDigest instance. The solution here reinstates the original, correct usage.  So technically, we're not working around the JDK bug. Of course, the bug symptom will be addressed in the pertinent support documentation.